sklearn/neighbors/tests/test_graph.py

```diff
@@ -81,6 +84,52 @@ def test_explicit_diagonal():
     assert not _has_explicit_diagonal(X2t)
 
 
+@pytest.mark.parametrize(
+    "params, expected_exception, msg",
+    [
+        (
+            {"p": 0},
+            ValueError,
+            "The 'p' parameter of kneighbors_graph must be a float in the range",
+        ),
+        (
+            {"p": 0.5},
+            UserWarning,
+            "Mind that for 0 < p < 1, Minkowski metrics are not distance metrics",
+        ),
+        (
+            {"p": 1},
+            None,
+            "No exception or warning expected",
+        ),
+        (
+            {"p": 5.7},
+            None,
+            "No exception or warning expected",
+        ),
+        (
+            {"p": float("inf")},
+            None,
+            "No exception or warning expected",
+        ),
```

Comment:
We usually don't check that no errors or warning are raised when we don't expect that
```suggestion
```